src/main/java/teammates/ui/controller/InstructorFeedbacksPageData.java

```diff
@@ -1,47 +1,354 @@
+                                                                             existingFeedbackSessions, instructors, 
+                                                                             feedbackSessionNameForSessionList, courseIdForNewSession);
+        fsList = new FeedbackSessionsTable(existingFeedbackSessionsRow);
+    }
+
+    private void buildNewForm(List<CourseAttributes> courses, String courseIdForNewSession,
+                              Map<String, InstructorAttributes> instructors,
+                              FeedbackSessionAttributes newFeedbackSession, String feedbackSessionType,
+                              String feedbackSessionNameForSessionList) {
+        
+        List<String> courseIds = new ArrayList<String>();
         for (CourseAttributes course : courses) {
+            courseIds.add(course.id);
+        }
+        
+        
+        newFsForm = buildBasicForm(courses, courseIdForNewSession, instructors, 
+                                   newFeedbackSession, feedbackSessionType,
+                                   feedbackSessionNameForSessionList, courseIds);
+        
+        AdditionalSettingsFormSegment additionalSettings = buildFormAdditionalSettings(newFeedbackSession);
+        newFsForm.setAdditionalSettings(additionalSettings);                  
+        
+    }
+
+    private FeedbackSessionsForm buildBasicForm(List<CourseAttributes> courses, String courseIdForNewSession,
+                                                Map<String, InstructorAttributes> instructors,
+                                                FeedbackSessionAttributes newFeedbackSession, String feedbackSessionType,
+                                                String feedbackSessionNameForSessionList, List<String> courseIds) {
+        FeedbackSessionsForm newFsForm = new FeedbackSessionsForm();
+        newFsForm.setCourseIdForNewSession(courseIdForNewSession);
+        
+        newFsForm.setFsName(newFeedbackSession == null ? "" : newFeedbackSession.feedbackSessionName);
+        
+        newFsForm.setCourses(courseIds);
+        
+        newFsForm.setFeedbackSessionTypeOptions(getFeedbackSessionTypeOptions(feedbackSessionType));
+
+        newFsForm.setFeedbackSessionNameForSessionList(feedbackSessionNameForSessionList);
+        
+        newFsForm.setCoursesSelectField(getCourseIdOptions(courses,  courseIdForNewSession, 
+                                                           instructors, newFeedbackSession));
+        
+        newFsForm.setTimezoneSelectField(getTimeZoneOptionsAsElementTags(newFeedbackSession));
+        
+        
+        newFsForm.setInstructions(newFeedbackSession == null ?
+                                  "Please answer all the given questions." :
+                                  sanitizeForHtml(newFeedbackSession.instructions.getValue()));
+        
+        newFsForm.setFsStartDate(newFeedbackSession == null ?
```

Comment:
I'm not sure if it will help, but do you think a single `if` block for those options that rely on the nullity of `newFeedbackSession` would be more neat?
````
a = z ? c : d;
b = z ? e : f;

versus

if (z) {
    a = c;
    b = e;
} else {
    a = d;
    b = f;
}
````
I think it is fine as it is, but go ahead and try it out if you want.